src/Memory/BotState.ts

```diff
@@ -66,17 +63,21 @@ export enum BotStateType {
 
 export class BotState {
     private static _instance: BotState | undefined
+    private readonly conversationReferenceToConversationIdMapper: (ref: Partial<BB.ConversationReference> | null) => string | null
     public memory: CLMemory | undefined
 
-    private constructor(init?: Partial<BotState>) {
+    private constructor(init?: Partial<BotState>,
+        conversationReferenceToConvIdMapper: (ref: Partial<BB.ConversationReference> | null) => string | null = BotState.DefaultConversationIdMapper) {
```

Comment:
Minor, but could extract this out to separate type so it could be re-used.

```
type ConvIdMapper = (ref: Partial<BB.ConversationReference> | null) => string | null
```